src/SFA.DAS.Roatp.CourseManagement/Application/Standards/Queries/GetProviderCourseQuery/GetProviderCourseQueryHandler.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Roatp.CourseManagement.InnerApi.Requests;
+using SFA.DAS.Roatp.CourseManagement.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.Roatp.CourseManagement.Application.Standards.Queries.GetProviderCourseQuery
+{
+    public class GetProviderCourseQueryHandler : IRequestHandler<GetProviderCourseQuery,GetProviderCourseResult>
+    {
+        private readonly ICoursesApiClient<CoursesApiConfiguration> _coursesApiClient;
+        private readonly IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> _courseManagementApiClient;
+        private readonly ILogger<GetProviderCourseQueryHandler> _logger;
+
+        public GetProviderCourseQueryHandler(IRoatpCourseManagementApiClient<RoatpV2ApiConfiguration> courseManagementApiClient, ICoursesApiClient<CoursesApiConfiguration> coursesApiClient,ILogger<GetProviderCourseQueryHandler> logger)
+        {
+            _coursesApiClient = coursesApiClient;
+            _courseManagementApiClient = courseManagementApiClient;
+            _logger = logger;
+        }
+
+        public async Task<GetProviderCourseResult> Handle(GetProviderCourseQuery request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Get Provider Course request received for ukprn {ukprn}, LarsCode {larsCode}", request.Ukprn, request.LarsCode);
+
+            GetStandardResponse standard;
+            try
```

Comment:
Remove the higher level try catch, this should be handled by a middleware

Reply:
I thought we'd established the middleware in apim doesn't do this, so we have to leave this in.  I'll dive in and have a look